### PR TITLE
feat(ingest): Starburst Trino usage

### PR DESCRIPTION
**Pull request by treff7es**
Starburst Trino usage

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-ingestion/source_docs/trino.md [hunk 1]
```diff
@@ -69,6 +69,65 @@ As a SQL-based service, the Trino integration is also supported by our SQL profi
...
+| `top_n_queries`        |          | `10`                                                           | Number of top queries to save to each table.                                                                                                                                                                                                                                                                                                                                           |
+| `extra_client_options` |          |                                                                | Additional options to pass to `google.cloud.logging_v2.client.Client`.                                                                                                                                                                                                                                                                                                                 |
```
Comment (shirshanka) [inline metadata-ingestion/source_docs/trino.md:127]: is this a valid config?

File: metadata-ingestion/source_docs/trino.md [hunk 2]
```diff
@@ -69,6 +69,65 @@ As a SQL-based service, the Trino integration is also supported by our SQL profi
...
+| `query_log_delay`     |          |                                                                | To account for the possibility that the query event arrives after the read event in the audit logs, we wait for at least `query_log_delay` additional events to be processed before attempting to resolve BigQuery job information from the logs. If `query_log_delay` is `None`, it gets treated as an unlimited delay, which prioritizes correctness at the expense of memory usage. |
```
Comment (shirshanka) [inline metadata-ingestion/source_docs/trino.md:128]: reference to BigQuery here

File: metadata-ingestion/src/datahub/ingestion/source/usage/starburst_trino_usage.py [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+                if not event_dict["accessed_metadata"]:
+                    logging.info("Field accessed_metadata is empty. Skipping ....")
+                    continue
+
+                event_dict["accessed_metadata"] = json.loads(
+                    event_dict["accessed_metadata"]
+                )
+
+                if not event_dict.get("usr") or event_dict["usr"] == "":
```
Comment (shirshanka) [inline metadata-ingestion/src/datahub/ingestion/source/usage/starburst_trino_usage.py:181]: can simplify with `if not event_dict.get("usr"):`  
since `if not ""` evaluates to True.

File: metadata-ingestion/src/datahub/ingestion/source/usage/starburst_trino_usage.py [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+                    agg_bucket = datasets[floored_ts].setdefault(
+                        resource,
+                        AggregatedDataset(
+                            bucket_start_time=floored_ts, resource=resource
+                        ),
+                    )
+
+                    # add @unknown.com to username
+                    # current limitation in user stats UI, we need to provide email to show users
+                    username = f"{event.usr if event.usr else 'unknown'}@{self.config.email_domain}"
```
Comment (shirshanka) [inline metadata-ingestion/src/datahub/ingestion/source/usage/starburst_trino_usage.py:223]: should only add `@email_address` if `event.usr` does not contain an `@`

File: metadata-ingestion/src/datahub/ingestion/source/sql/sql_common.py [hunk 5]
```diff
@@ -418,9 +418,17 @@ def loop_tables(
                 self.report.report_dropped(dataset_name)
                 continue
 
-            columns = inspector.get_columns(table, schema)
-            if len(columns) == 0:
-                self.report.report_warning(dataset_name, "missing column information")
+            try:
```
Comment (treff7es) [inline metadata-ingestion/src/datahub/ingestion/source/sql/sql_common.py:421]: @shirshanka should I remove this or it can stay here. 
It make sure for sources to handle gracefully if there is a problem parsing a column. This does not belong strictly to this pr, so I think maybe this should be in a separate pr if you are ok with it.
This helps to not restart the whole source from the beginning if it turns out there is an invalid column information.

**Review by shirshanka — APPROVED** [review]
LGTM!

File: metadata-ingestion/src/datahub/ingestion/source/sql/sql_common.py (hunk 5 above)
Comment (shirshanka) [reply metadata-ingestion/src/datahub/ingestion/source/sql/sql_common.py:421]: lgtm
